### ISSUE TITLE
Provide containerd Windows default CRI endpoint

### ISSUE THREAD

**Issue by meshuga** [rancher-sandbox/rancher-desktop#2254]
### Problem Description

As of today, requests to containerd are invoked through nerdctl proxy, that uses wsl.exe to invoke commands within `rancher-desktop` runtime.

It makes it unusable to use containerd client (https://github.com/containerd/containerd/blob/v1.6.4/client.go) and all tools that use this SDK.

### Proposed Solution

I propose to write an npipe proxy, to allow connectivity on default address `npipe://.//pipe//containerd-containerd`.

Similar solution was already created for Docker, see https://github.com/rancher-sandbox/rancher-desktop/blob/v1.3.0/src/go/wsl-helper/pkg/dockerproxy/platform/serve_windows.go

**Comment by jandubois** [comment]
> As of today, requests to containerd are invoked through nerdctl proxy, that uses wsl.exe to invoke commands within `rancher-desktop` runtime.

This will still need to happen, even if the containerd socket it forwarded to the Win32 side because `nerdctl` needs access to the image storage in order to do `nerdctl build`.

This is the first request we received for this functionality; can you give some examples of native Windows applications that would talk to the containerd socket directly? I realize that this may be a chicken-and-egg problem, but am curious if there is already something out there.

**Comment by meshuga** [comment]
I was thinking of writing a tool for writing integration tests, similar to https://www.testcontainers.org/ for containerd. The project maintainers are reluctant to provide containerd support (see testcontainers/testcontainers-java#4664)

With that in mind, I can’t just make use of the containerd client when using Rancher Desktop.

**Comment by jandubois** [comment]
> With that in mind, I can’t just make use of the containerd client when using Rancher Desktop.

You mean via `ctr` or `crictl`? Is there anything those can do that you can't do with `nerdctl`?

Having said that, you should be able to use `ctr` and `crictl` from within WSL2 distros. Hmm, we may not bind the socket into other distros, like we do for the docker socket, but at least that should be easy to implement. So right now it may only work from inside the rancher-desktop distro.

Would having the socket available inside other WSL2 distro help you? I'm not saying that we shouldn't/wouldn't do the named pipe thing, but that may take longer until it can be scheduled.

**Comment by meshuga** [comment]
The client I mentioned doesn't use the commands since it is being used by those commands you mentioned. I may use the commands directly but I think it should be Rancher Desktop responsibility to implement the expected interface.

**Comment by wu-yafeng** [comment]
Does any progress on this issue?

**Comment by jandubois** [comment]
> Does any progress on this issue?

No, there is no progress. There has also not been any request for this beyond the original poster here, so this is unlikely to get prioritized any time soon.

Note that Rancher Desktop is an open source project, so you can always contribute functionality that is important to you yourself. Do talk with us first though, to make sure any proposed architectural changes are acceptable to us!
